### PR TITLE
MAINT: Clean up pandas/util/testing.py

### PR DESCRIPTION
**Pull request by gfyoung**
Transform testing methods to use more `pytest` idiom.

xref #15990.

### CODE REVIEW

File: pandas/tests/groupby/test_timegrouper.py [hunk 1]
```diff
@@ -571,8 +571,7 @@ def test_groupby_with_timezone_selection(self):
     def test_timezone_info(self):
         # GH 11682
         # Timezone info lost when broadcasting scalar datetime to DataFrame
-        tm._skip_if_no_pytz()
-        import pytz
+        pytz = pytest.importorskip("pytz")
```
Comment (jreback) [inline pandas/tests/groupby/test_timegrouper.py]: actually we should remove this entirely
we require pytz (prob lots of skips related to pytz in the codebase - should remove all of them)

File: pandas/tests/indexes/datetimes/test_construction.py [hunk 2]
```diff
@@ -376,7 +376,7 @@ def test_constructor_datetime64_tzformat(self):
                                      tz='Asia/Tokyo')
             tm.assert_numpy_array_equal(idx.asi8, expected_i8.asi8)
 
-        tm._skip_if_no_dateutil()
+        pytest.importorskip("dateutil")
```
Comment (jreback) [inline pandas/tests/indexes/datetimes/test_construction.py]: these are also required so again can remove

Comment (gfyoung) [inline pandas/tests/indexes/datetimes/test_construction.py]: Okay, so just assume we have `dateutil` installed?

File: pandas/tests/groupby/test_timegrouper.py (hunk 1 above)
Comment (gfyoung) [inline pandas/tests/groupby/test_timegrouper.py]: Okay, so just assume we have `pytz` installed?

Comment (jreback) [inline pandas/tests/groupby/test_timegrouper.py]: see if anything breaks when these are installed

File: pandas/tests/indexes/datetimes/test_construction.py (hunk 2 above)
Comment (jreback) [inline pandas/tests/indexes/datetimes/test_construction.py]: see if anything breaks when these are installed

**Comment by gfyoung** [comment]
@jreback : Not sure what `codecov` is complaining about.  However, I made all of the requested changes, and everything else is green.

**Comment by jreback** [comment]
@gfyoung yeah I am about ready to disable codecov. It is pretty bogus. I haven't been able to make it use the latest master as the comparison (not really sure why) and their help services are not very helpful :< 

if you want to have a look at other code coverage services we could easily switch.

**Comment by jreback** [comment]
thanks!

**Pull request merged**